go/mysql/conn.go

```diff
@@ -536,24 +443,11 @@ func (c *Conn) startEphemeralPacket(length int) []byte {
 		panic("startEphemeralPacket cannot be used while a packet is already started.")
 	}
 
-	// Fast path: we can reuse a single memory buffer for
-	// both the header and the data.
-	if length <= cap(c.buffer)-4 {
-		c.currentEphemeralPolicy = ephemeralWriteGlobalBuffer
-		c.buffer = c.buffer[:length+4]
-		c.buffer[0] = byte(length)
-		c.buffer[1] = byte(length >> 8)
-		c.buffer[2] = byte(length >> 16)
-		c.buffer[3] = c.sequence
-		c.sequence++
-		return c.buffer[4:]
-	}
-
-	// Slower path: we can use a single buffer for both the header and the data, but it has to be allocated.
+	// get buffer from pool
 	if length < MaxPacketSize {
```

Comment:
you shouldn't need this length check anymore. same with read.

and also BigBuffer can go away (or whatever the last "policy" is)

Reply:
True for writes.
but for the reads code is different for large buffers - it reads packets one by one instead of ReadAll.

Reply:
ah okay. feel free to leave as is